ext/xmlreader/php_xmlreader.stub.php

```diff
@@ -71,11 +71,11 @@ public function setParserProperty(int $property, bool $value) {}
     public function setRelaxNGSchema(?string $filename) {}
 
     /** @return bool */
-    public function setRelaxNGSchemaSource(?string $source) {}
+    public function setRelaxNGSchemaSource(?string $value) {}
 
     /** @return bool|XMLReader */
-    public static function XML(string $source, ?string $encoding = null, int $options = 0) {}
+    public static function XML(string $value, ?string $encoding = null, int $flags = 0) {}
 
     /** @return DOMNode|false|null */
-    public function expand(?DOMNode $basenode = null) {}
+    public function expand(?DOMNode $document = null) {}
```

Comment:
I'm not sure we can call this parameter `$document` (mainly because I'm not familar with ext/xmlreader)

Reply:
Off: If ext/dom is not available, `XMLReader::expand()` triggers a warning and returns `null` immediately, which whacks ZPP handling... That said, shouldn't we define the function itself conditionally instead?

Reply:
> I'm not sure we can call this parameter `$document` (mainly because I'm not familar with ext/xmlreader)

I wanted to call it `$document` because it is typically a `DOMDocument` that the created node will be associated with. It accepts more than just `DOMDocument` instances though, so maybe `$baseNode` is a better idea.

> Off: If ext/dom is not available, XMLReader::expand() triggers a warning and returns null immediately, which whacks ZPP handling... That said, shouldn't we define the function itself conditionally instead?

Wow, I did not notice that before. What exactly do you mean with "whacks ZPP handling"? Are you referring to the parameters not being parsed and checked?

Having the function disappear completely also does not sound ideal. I think I would rather have it throwing an exception that clearly states what the problem is. This problem also makes me wonder why we have the possibility to compile PHP without DOM support in the first place. Dropping that possibility would be the simplest solution.

Reply:
I've addressed the zpp issue in https://github.com/php/php-src/commit/faea5ab837ab6393c8821f85cf8abe2723593e8e by making the function throw rather than warning and returning null.

> This problem also makes me wonder why we have the possibility to compile PHP without DOM support in the first place. Dropping that possibility would be the simplest solution.

This is impossible, because dom requires libxml.

Reply:
... and libxml is not available on some of the supported platforms?

Reply:
> ... and libxml is not available on some of the supported platforms?

That's unlikely. However, a minimal PHP build should not have any dependencies beyond libc. Any dependencies a minimal PHP build needs have to be vendored, and we certainly wouldn't want to vendor libxml.

Reply:
Ah, ok. I am not familiar with minimal PHP builds. Anyways, thanks for fixing the ZPP issue!